pkg/store/prometheus.go

```diff
@@ -496,9 +499,32 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 		return &storepb.LabelValuesResponse{Values: []string{l}}, nil
 	}
 
-	vals, err := p.client.LabelValuesInGRPC(ctx, p.base, r.Label, nil, r.Start, r.End)
-	if err != nil {
-		return nil, err
+	var (
+		vals []string
+		sers []map[string]string
+		err  error
+	)
+
+	version := p.promVersion()
+	if version > "2.24" {
+		vals, err = p.client.LabelValuesInGRPC(ctx, p.base, r.Label, r.Matchers, r.Start, r.End)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		matchers, err := storepb.MatchersToPromMatchers(r.Matchers...)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		sers, err = p.client.SeriesInGRPC(ctx, p.base, matchers, r.Start, r.End)
+		if err != nil {
+			return nil, err
+		}
+		for _, s := range sers {
+			if val, exists := s[r.Label]; exists {
```

Comment:
@Namanl2001, please take a look at https://github.com/thanos-io/thanos/blob/main/pkg/api/query/v1.go#L870-L886.
Here `s.[r.Label]` might get duplicate values so you need to use a set to deduplicate first and then convert it to a slice later.

Reply:
Addressed!